VectorInterpolator.C

```diff
@@ -0,0 +1,164 @@
+                 rotation_matrices,
+                 ref_point,
+                 epsilon)
+{
+    CAROM_VERIFY(reduced_vectors.size() == rotation_matrices.size());
+
+    // Rotate the reduced vectors
+    for (int i = 0; i < reduced_vectors.size(); i++)
+    {
+        // The ref_point does not need to be rotated
+        if (i == d_ref_point)
+        {
+            d_rotated_reduced_vectors.push_back(reduced_vectors[i]);
+        }
+        else
+        {
+            Vector* Q_tA = rotation_matrices[i]->transposeMult(reduced_vectors[i]);
+            d_rotated_reduced_vectors.push_back(Q_tA);
+        }
+    }
+}
+
+void VectorInterpolator::obtainLambda(std::vector<Vector*> gammas)
+{
+
+    // Solving f = B*lambda
+    Matrix* f_T = new Matrix(gammas[0]->dim(), gammas.size(), false);
+    for (int i = 0; i < f_T->numRows(); i++)
+    {
+        for (int j = 0; j < f_T->numColumns(); j++)
+        {
+            f_T->item(i, j) = gammas[j]->getData()[i];
+        }
+    }
+
+    // Obtain B vector by calculating inverse quadratic distance.
+    Matrix* B = new Matrix(gammas.size(), gammas.size(), false);
+    for (int i = 0; i < B->numRows(); i++)
+    {
+        for (int j = i + 1; j < B->numColumns(); j++)
+        {
+            Vector diff;
+            d_parameter_points[i]->minus(*d_parameter_points[j], diff);
+            double res = 1.0 / (1.0 + std::pow(diff.norm(), 2));
+            B->item(i, j) = res;
+            B->item(j, i) = res;
+        }
+    }
+
+    int gamma_size = gammas.size();
+    int num_elements = gammas[0]->dim();
+    int* ipiv = new int[gamma_size];
+    int info;
+
+    dgesv(&gamma_size, &num_elements, B->getData(),  &gamma_size, ipiv, f_T->getData(), &gamma_size, &info);
+    CAROM_VERIFY(info == 0);
+
+    delete [] ipiv;
+    delete B;
+
+    d_lambda_T = f_T;
+}
+
+Vector* VectorInterpolator::obtainLogInterpolatedVector(std::vector<double> inv_q)
+{
+    Vector* log_interpolated_vector = new Vector(d_rotated_reduced_vectors[d_ref_point]->dim(), d_rotated_reduced_vectors[d_ref_point]->distributed());
+    CAROM_VERIFY(d_rotated_reduced_vectors[d_ref_point]->dim() == d_lambda_T->numRows());
+    for (int i = 0; i < d_lambda_T->numRows(); i++)
+    {
+        for (int j = 0; j < inv_q.size(); j++)
+        {
+            log_interpolated_vector->getData()[i] += d_lambda_T->item(i, j) * inv_q[j];
+        }
+    }
+
+    return log_interpolated_vector;
+}
+
+Vector* VectorInterpolator::interpolate(Vector* point)
+{
+    if (d_lambda_T == NULL)
+    {
+        // Perform log mapping by doing Y - X.
+        std::vector<Vector*> gammas;
+
+        for (int i = 0; i < d_parameter_points.size(); i++)
+        {
+            // For ref point, gamma is the zero vector.
+            if (i == d_ref_point)
+            {
+                Vector* gamma = new Vector(d_rotated_reduced_vectors[d_ref_point]->dim(), d_rotated_reduced_vectors[d_ref_point]->distributed());
+                gammas.push_back(gamma);
```

Comment:
Is `gamma = 0.0;` necessary?

Reply:
I talked to Youngsoo about this. It seems it is required. Perhaps he can expand. @chldkdtn

Reply:
I agree that `gamma` should be zero, but my question is whether the constructor initializes it to zero (as in the discussion about the `Matrix` constructor.